freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UmsUsersStateProvider.java

```diff
@@ -241,12 +258,14 @@ private FmsUser umsMachineUserToUser(MachineUser umsMachineUser) {
         // TODO what should the appropriate first and last name be for machine users?
         fmsUser.withFirstName("Machine");
         fmsUser.withLastName("User");
+        fmsUser.withAzureObjectId(getOptionalAzureObjectId(umsMachineUser.getCloudIdentitiesList()));
         return fmsUser;
     }
 
     private FmsGroup umsGroupToGroup(Group umsGroup) {
         FmsGroup fmsGroup = new FmsGroup();
         fmsGroup.withName(umsGroup.getGroupName());
+        fmsGroup.withAzureObjectId(getOptionalAzureObjectId(umsGroup.getCloudIdentitiesList()));
```

Comment:
FmsGroups suffer the same issue as FmsUsers.
you'll need to calculate the group mapping somewhere around line 71 and add it into the UmsUsersState somewhere around line 78